sql/opt_range.h

```diff
@@ -1651,6 +1651,39 @@ class SQL_SELECT :public Sql_alloc {
 };
 
 
+class SQL_SELECT_auto
+{
+  SQL_SELECT *select;
+public:
+  SQL_SELECT_auto(): select(NULL)
+  {}
+  ~SQL_SELECT_auto()
+  {
+    if (select)
```

Comment:
This check is redundant: it's safe to pass `NULL` to `delete`.

Reply:
Fixed.